### PR TITLE
[Merged by Bors] - feat(bench_summary): report no significant changes and job_id

### PR DESCRIPTION
**Pull request by adomani**
If the `!bench` summary contains no entry with a difference in instructions of at least 10^9, then the bot posts a message confirming this information, rather than silently failing.

The bot also posts a link to the workflow CI run that generated the message.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by adomani** [comment]
Here are the [benchmark results](http://speed.lean-fro.org/mathlib4/run-detail/1319cc1e-dff9-4af9-920e-00ac47e5f39b) for commit b559fb5c8dd7c4c36b42167109a210483c8e095b.
There were [no significant changes](http://speed.lean-fro.org/mathlib4/compare/6472c384-e9cd-462f-b8f2-c0f372ad374b/to/1319cc1e-dff9-4af9-920e-00ac47e5f39b) against commit b696b3524d2dde1068fa8192e92f25c990e9eb15.

**Comment by adomani** [comment]
Here are the [benchmark results](http://speed.lean-fro.org/mathlib4/run-detail/1319cc1e-dff9-4af9-920e-00ac47e5f39b) for commit b559fb5c8dd7c4c36b42167109a210483c8e095b.
There were [no significant changes](http://speed.lean-fro.org/mathlib4/compare/6472c384-e9cd-462f-b8f2-c0f372ad374b/to/1319cc1e-dff9-4af9-920e-00ac47e5f39b) against commit b696b3524d2dde1068fa8192e92f25c990e9eb15.

**Comment by adomani** [comment]
Here are the [benchmark results](http://speed.lean-fro.org/mathlib4/run-detail/1319cc1e-dff9-4af9-920e-00ac47e5f39b) for commit b559fb5c8dd7c4c36b42167109a210483c8e095b.
There were [no significant changes](http://speed.lean-fro.org/mathlib4/compare/6472c384-e9cd-462f-b8f2-c0f372ad374b/to/1319cc1e-dff9-4af9-920e-00ac47e5f39b) against commit b696b3524d2dde1068fa8192e92f25c990e9eb15.

**Comment by adomani** [comment]
Here are the [benchmark results](http://speed.lean-fro.org/mathlib4/run-detail/1319cc1e-dff9-4af9-920e-00ac47e5f39b) for commit b559fb5c8dd7c4c36b42167109a210483c8e095b.
There were [no significant changes](http://speed.lean-fro.org/mathlib4/compare/6472c384-e9cd-462f-b8f2-c0f372ad374b/to/1319cc1e-dff9-4af9-920e-00ac47e5f39b) against commit b696b3524d2dde1068fa8192e92f25c990e9eb15.

**Comment by adomani** [comment]
No benchmark entry differed by at least 1000000000 instructions

**Comment by adomani** [comment]
No benchmark entry differed by at least 1000000000 instructions

**Comment by adomani** [comment]
No benchmark entry differed by at least +1.0⬝10⁹ instructions

**Comment by adomani** [comment]
No benchmark entry differed by at least +1.0⬝10⁹ instructions.

**Comment by adomani** [comment]
Here are the [benchmark results](http://speed.lean-fro.org/mathlib4/run-detail/1319cc1e-dff9-4af9-920e-00ac47e5f39b) for commit b559fb5c8dd7c4c36b42167109a210483c8e095b.
There were [no significant changes](http://speed.lean-fro.org/mathlib4/compare/6472c384-e9cd-462f-b8f2-c0f372ad374b/to/1319cc1e-dff9-4af9-920e-00ac47e5f39b) against commit b696b3524d2dde1068fa8192e92f25c990e9eb15.

**Comment by adomani** [comment]
No benchmark entry differed by at least +1.0⬝10⁹ instructions.

File: scripts/bench_summary.lean [hunk 1]
```diff
@@ -197,7 +199,8 @@ def addBenchSummaryComment (PR : Nat) (repo : String)
   IO.FS.writeFile (tempFile ++ ".src") bench
 
   -- Extract all instruction changes whose magnitude is larger than `threshold`.
-  let threshold := s!"{10 ^ 9}"
+  let thr := 10 ^ 9
+  let threshold := s!"{thr}"
```
Comment (joneugster) [inline scripts/bench_summary.lean:203]: optional: if I see this correctly and this is indeed just used once below (I think the second time you use `{formatDiff thr`}` instead) then you could remove the var `threshold` and rename the current `thr : Nat` into `threshold` instead.

**Review by joneugster — APPROVED** [review]

**Comment by joneugster** [comment]
LGTM! Thank you

maintainer delegate

File: scripts/bench_summary.lean (hunk 1 above)
Comment (adomani) [reply scripts/bench_summary.lean]: Thanks Jon!

My intention had bee to have a single entry point for the `Nat`-valued `threshold`, but I was modifying code that already had a single entry to the `String`-valued one, hence the awkwardness!  I revised the code following your suggestion, so now `threshold` really is the single `Nat`-valued "variable" and the `String` version is in-lined: I am happy with the change!

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**